app/policies/ownership_policy.rb

```diff
@@ -0,0 +1,15 @@
+class OwnershipPolicy < ApplicationPolicy
+  class Scope < Scope
+    def resolve
+      scope.confirmed.or(scope.where(rubygem: user.rubygems)).or(scope.where(user: user))
```

Comment:
This could easily be potentially impactful on performance if someone owns a lot of gems. I'm wondering what benefit applying a scope here really gives us?

Reply:
@martinemde can you post what query this resolves to? I _think_ it should collapse to a single left join on an index, so that should be OK

Reply:
For the curious, even though I removed it:

```sql
SELECT "ownerships".* FROM "ownerships"
WHERE ("ownerships"."confirmed_at" IS NOT NULL OR "ownerships"."rubygem_id" IN (
  SELECT "rubygems"."id"
  FROM "rubygems"
  INNER JOIN "ownerships" ON "rubygems"."id" = "ownerships"."rubygem_id"
  WHERE "ownerships"."user_id" = $1 AND "ownerships"."confirmed_at" IS NOT NULL
) OR "ownerships"."user_id" = $2)
```

If we use scopes, we'll probably need to be more careful. That could be optimized at the very least to use rubygem_id from the ownerships table instead of loading the association.